src/utils/registerNewNetwork.ts

```diff
@@ -74,38 +75,75 @@ export const registerNewNetwork = async (
     blockTime: arbitrumSdkConstants.ARB_MINIMUM_BLOCK_TIME_IN_SECONDS,
   };
 
-  let isL1NetworkRegistered = false;
-  try {
-    await getL1Network(l1Provider);
-    isL1NetworkRegistered = true;
-  } catch {
-    isL1NetworkRegistered = false;
-  }
+  // If the parent chain is an Arbitrum chain, we need to register parent and child chain both as L2 chains
+  if (await isArbitrumChain(parentProvider)) {
+    const isParentNetworkRegistered = await isNetworkRegistered(parentProvider, { type: 'L2' });
```

Comment:
I'd maybe move the construction of these `L1Network` and `L2Network` objects outside of this function so it's cleaner to read